### PR TITLE
separating docker compose for pmml and mlflow

### PR DESCRIPTION
**Pull request by Jadw1**
create_environment.sh can now take arguments. By default, with no arguments it creates environment with all frameworks. However you can specify framework to launch e.g. `./create_environment.sh mlflow` will create environment with only mlflow framework.

`./create_environment.sh mlflow pmml` is equal to `./create_environment.sh`

There is also parameter `-e` that stops nussknacker from running and doesn't compile prinz.

However, current solution has one downside. Both compopse files for pmml and mlflow have separated proxy containers.

### CODE REVIEW

File: dev-environment/create_environment.sh [hunk 1]
```diff
@@ -2,21 +2,46 @@
...
+        elif [ $ARG = "pmml" ]
+        then
+            COMP_FILES+="-f docker-compose-pmml.yaml "
+        fi
+    done
+fi
+
+if [ $COMPILE = true ]
+then
+    echo "COOOOOMPILE"
```
Comment (kantoniak) [inline dev-environment/create_environment.sh]: Leftovers? 😄

**Comment by kantoniak** [comment]
I didn't dive into the code too much, but building only required services _may_ solve the problem you mentioned.

File: dev-environment/clear_environment.sh [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+    COMP_FILES="-f docker-compose-mlflow.yaml -f docker-compose-pmml.yaml"
+else
+    for FRAMEWORK in $@
+    do
+        if [ $FRAMEWORK = "mlflow" ]
+        then
+            COMP_FILES+="-f docker-compose-mlflow.yaml "
+        elif [ $FRAMEWORK = "pmml" ]
+        then
+            COMP_FILES+="-f docker-compose-pmml.yaml"
```
Comment (avan1235) [inline dev-environment/clear_environment.sh]: space missing here imo at the end of string

File: .env [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 MODEL_1_PORT=1111
 MODEL_2_PORT=2222
 MODEL_3_PORT=3333
 MLFLOW_SERVER_PORT=5000
 POSTGRES_PORT=5432
 NGINX_BUCKET_PORT=9000
 
-NGINX_STATIC_PORT=4242
-
 PMML_SAMPLES_PORT=5100
```
Comment (avan1235) [inline .env:10]: it is not for PMML (as I see you changed the name to NGINX_PMML_STATIC_PORT) - it is serving extra parametrs for model that are not provided for nussknacker and which should be downloaded additionally - it should be in something like 'abstract' docker comopse which can be used for mlfloe as well as for pmml as it is implemented in prinz api (not in pmml nor in mlflow)

File: dev-environment/create_environment.sh (hunk 1 above)
Comment (avan1235) [reply dev-environment/create_environment.sh]: to be removed

File: dev-environment/docker-compose-pmml.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      - pmml-net
+
+  proxy:
+    image: nginx:alpine
+    restart: always
+    volumes:
+      - ./nginx/pmml/templates:/etc/nginx/templates
+      - ./nginx/pmml/static:/etc/nginx/html
+    ports:
+      - "${NGINX_PMML_STATIC_PORT}:${NGINX_PMML_STATIC_PORT}"
```
Comment (avan1235) [inline dev-environment/docker-compose-pmml.yaml]: as described - it shouldn't be here as it is not created for pmml

File: dev-environment/nginx/pmml/templates/default.conf.template [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    location /static/double {
+        rewrite /static/double /double break;
+    }
+    location /static/boolean {
+        rewrite /static/boolean /boolean break;
+    }
+    location /static/string {
+        rewrite /static/string /string break;
+    }
+}
```
Comment (avan1235) [inline dev-environment/nginx/pmml/templates/default.conf.template]: this is not meant to be served for pmml - we use this data to test the model proxy which can add extra parameters to model so it is should be separated

**Review by avan1235 — CHANGES_REQUESTED** [review]

**Review by avan1235 — APPROVED** [review]

**Pull request merged**